File: 02_ProgrammingFundamentals/_02_DataTypesAndVariables/_02_Exercise/_10_PokeMon.java
package _02_ProgrammingFundamentals._02_DataTypesAndVariables._02_Exercise;

import java.util.Scanner;

public class _10_PokeMon {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int pokemonPowerN = Integer.parseInt(scan.nextLine());
        int distanceM = Integer.parseInt(scan.nextLine());
        int exhaustionFactorY = Integer.parseInt(scan.nextLine());

        // Add variables
        int originalValue = pokemonPowerN / 2;
        int successfulPokes = 0;

        // Check if pokemon can make the move
        while(pokemonPowerN >= distanceM){
            successfulPokes++;
            pokemonPowerN -= distanceM;
            
            if (pokemonPowerN == originalValue && exhaustionFactorY != 0) {
                pokemonPowerN /= exhaustionFactorY;
            }
        }

        // Print result
        System.out.println(pokemonPowerN);
        System.out.println(successfulPokes);

        scan.close();
    }
}

File: 24_C#Fundamentals/_02_DataTypesAndVariables/_02_Exercise/_10_PokeMon/_10_PokeMon.cs
using System;

public class _10_PokeMon
{
    public static void Main(string[] args)
    {
        // Read input values
        int pokemonPowerN = int.Parse(Console.ReadLine());
        int distanceM = int.Parse(Console.ReadLine());
        int exhaustionFactorY = int.Parse(Console.ReadLine());

        // Add variables
        int originalValue = pokemonPowerN / 2;
        int successfulPokes = 0;

        // Check if pokemon can make the move
        while (pokemonPowerN >= distanceM)
        {
            successfulPokes++;
            pokemonPowerN -= distanceM;

            if (pokemonPowerN == originalValue && exhaustionFactorY != 0)
            {
                pokemonPowerN /= exhaustionFactorY;
            }
        }

        // Print result
        Console.WriteLine(pokemonPowerN);
        Console.WriteLine(successfulPokes);
    }
}
